### ISSUE TITLE
Support static member variable in struct

### ISSUE THREAD

**Issue by antiagainst** [Microsoft/DirectXShaderCompiler#664]
For the following HLSL source code:

```hlsl
struct S
{
    static float4 data;
};

float4 S::data = 4.2;

float4 main() : SV_TARGET
{
    return S::data;
}
```

`fxc` happily accepts it but `dxc` errors out with

```
error: 'const' is not a valid modifier for a field
float4 S::data = 4.2;
^
```

Are we gonna support `static` struct member variable in `dxc`? Either way, the above error message is wrong since there is no `const` involved.

**Comment by python3kgae** [comment]
Young, the const is added by your https://github.com/Microsoft/DirectXShaderCompiler/commit/dc00d740281f350c1fcf1f519bd390822af23350.
Could you take a look?

**Comment by youngkim93** [comment]
This should be fixed with #668. For now we are making static data member to have internal linkage. This can be changed later after more discussions. Thanks!
